### PR TITLE
Show lambda term and word types in frontend

### LINKED ISSUE #9: Show lambda term and word types in frontend
The lambda term is accessible via `analyses[0].proof.term`. Basically, it is the part after the turnstile symbol ('⊢') of the line that you see at the bottom of the generated PDF (in red).

The words and their types can be extracted using `[(p.string, p.type) for p in analyses[0].lexical_phrases ]`. Æthel or Spindle should provide a method that turns the resulting objects into strings that can be sent to the frontend. There, they should be displayed in a simple table format.

These should be the main exports of a parse. Generating TeX and/or a PDF should only be available after an additional action/click.

### PR DESCRIPTION
**Pull request by bbonf**
Closes #9  

Some notes:
- The fact that on the frontend `parse()` uses `export()` seems quite backwards, but it made it very easy to add a new response type.
- It isn't ideal that everything is parsed from scratch on every export type (while the spindle-server already provides everything in one call), but in practice it doesn't matter much.
- The proof object passed around is also relevant for #10

### CODE REVIEW

**Review by XanderVertegaal — CHANGES_REQUESTED** [review]
Nice, well done! 👍 And thanks for laying out the groundwork for the proof export. That should be a breeze now :) I agree that it's fine for the analysis to run twice when you parse  + export. I don't think we should be worried about having too much traffic.

I have left some frontend comments. The caption for 'Term' and a table header are the most important; you may choose to implement/ignore the others as you see fit.

In addition, I noticed that parse results and export buttons remain visible after the parser returns an error. Maybe we could hide them instead (perhaps even already when the user changes the input). Feel free to make a separate issue out of this if you don't want to include that here, though.

File: frontend/src/app/spindle/spindle.component.html [hunk 1]
```diff
@@ -28,56 +28,86 @@ <h2 class="title" i18n>Spindle</h2>
...
-    <pre
-        name="tex-output"
-        class="is-family-code"
-        [innerText]="texOutput"
-        disabled
-    ></pre>
+<div *ngIf="term && lexicalPhrases">
+    <div class="block is-size-4">{{ term }}</div>
+    <table class="table">
+        <tr *ngFor="let phrase of lexicalPhrases; index as i">
```
[the remark this answers is not in the document]
Comment (bbonf) [reply frontend/src/app/spindle/spindle.component.html:44]: I'm happy to do that, but I'm not sure what the correct titles would be

File: frontend/src/app/spindle/spindle.component.html [hunk 2]
```diff
@@ -28,56 +28,86 @@ <h2 class="title" i18n>Spindle</h2>
...
+            <td>
+                {{ i }}
```
[the remark this answers is not in the document]
Comment (bbonf) [reply frontend/src/app/spindle/spindle.component.html:46]: The reason for using zero-based indexing is that these match the constants in the term (c0, c1, ...)

File: frontend/src/app/spindle/spindle.component.ts [hunk 3]
```diff
@@ -45,9 +49,18 @@ export class SpindleComponent implements OnInit, OnDestroy {
                 if (response.pdf) {
                     this.downloadPDF(response.pdf);
                 }
+                if (response.term && response.lexical_phrases) {
+                    this.term = response.term;
+                    this.lexicalPhrases = response.lexical_phrases;
+                    this.parsed = true;
```
[the remark this answers is not in the document]
Comment (bbonf) [reply frontend/src/app/spindle/spindle.component.ts:55]: This is indeed redundant

**Review by XanderVertegaal — APPROVED** [review]

**Pull request merged**